### PR TITLE
Fix translation doesn't work for child theme

### LINKED ISSUE #30716: The translation does not work for Child Theme
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

I make Child Theme from Classic Theme and upload it back. Set Child Theme as current. Next I want to make a translation in this Child Theme. The translation page is not loading.

### Expected behavior

Load with the possibility of translation or give an error.

### Steps to reproduce

1. Go to BO
2. Theme & Logo
3. Advanced Customization
4. Download theme child_classic.zip
5. Upload child theme child_classic.zip
6. Theme & Logo
7. Child theme of classic's theme - Use this theme
8. International -> Translations
9. Modify translations
10. Modify

CleanShot 2023-01-02 at 11 37 44@2x

### PrestaShop version(s) where the bug happened

1.7.8.8 and 8.0.0

### PHP version(s) where the bug happened

7.4.33

### If your bug is related to a module, specify its name and its version

_No response_

### PR DESCRIPTION
**Pull request by ThomasDrd**
Related to this issue https://github.com/PrestaShop/PrestaShop/issues/30716 Revert to previous version explained in this comment : https://github.com/PrestaShop/PrestaShop/issues/30716#issuecomment-1525153698

| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | Develop
| Description?      | Please be specific when describing the PR. <br> Every detail helps: versions, browser/server configuration, specific module/theme, etc. Feel free to add more information below this table.
| Type?             | bug fix
| Category?         |  BO 
| Deprecations?     | yes
| How to test?      | Changing core traduction applied in child theme
| Fixed ticket?     | Fixes #30716

### CODE REVIEW

**Comment by kpodemski** [comment]
Hi @ThomasDrd 

Thanks for the PR. I found a bit cleaner version of the fix, take a look:
https://capture.dropbox.com/ilxwKxmz4XRUzh7E

What do you think?

**Comment by arjan-designpro** [comment]
> Hi @ThomasDrd
> 
> Thanks for the PR. I found a bit cleaner version of the fix, take a look: https://capture.dropbox.com/ilxwKxmz4XRUzh7E
> 
> What do you think?

Does nothing for me.

**Comment by kpodemski** [comment]
@arjan-designpro have you cleared the cache? It does work for me.

**Comment by mariuszsienkiewicz** [comment]
Yes, it does work. If someone wants to test it, here is the updated content of the variable instead of the screenshot:

```php
$selectTranslationsQuery = '
    SELECT `key`, `translation`, `domain`
    FROM `' . _DB_PREFIX_ . 'translation`
    WHERE `id_lang` = ' . $localeResults[$locale]['id_lang'] . '
    AND theme ' . ($this->theme !== null ? ' IN ("' . $this->theme->getName() . '"' . ($this->theme->get('parent') ? ',"' . $this->theme->get('parent') . '"' : '') . ')' : 'IS NULL');
```

**Comment by tleon** [comment]
You also have some failing CI test to handle and maybe some refacto for the problem shown by the static analysis.

**Review by tleon — COMMENTED** [review]
You also have some failing CI test to handle and maybe some refacto for the problem shown by the static analysis.

**Comment by kpodemski** [comment]
Hi @ThomasDrd 

What do you think about using proposed solution?

**Comment by ThomasDrd** [comment]
Sorry for the response time.
I integrated the solution in my PR

File: src/PrestaShopBundle/Translation/Loader/SqlTranslationLoader.php [hunk 1]
```diff
@@ -74,16 +74,16 @@ public function load($resource, $locale, $domain = 'messages')
...
-            WHERE `id_lang` = ' . $localeResults[$locale]['id_lang'] . '
-            AND theme ' . ($this->theme !== null ? '= "' . $this->theme->getName() . '"' : 'IS NULL');
+                SELECT `key`, `translation`, `domain`
+                FROM `' . _DB_PREFIX_ . 'translation`
+                WHERE `id_lang` = ' . $localeResults[$locale]['id_lang'] . '
+                AND theme ' . ($this->theme !== null ? ' IN ("' . $this->theme->getName() . '"' . ($this->theme->get('parent') ? ',"' . $this->theme->get('parent') . '"' : '') . ')' : 'IS NULL');
 
         $translations = Db::getInstance()->executeS($selectTranslationsQuery) ?: [];
-
+
```
Comment (kpodemski) [inline src/PrestaShopBundle/Translation/Loader/SqlTranslationLoader.php]:
```suggestion
```

File: src/PrestaShopBundle/Translation/Loader/SqlTranslationLoader.php [hunk 2]
```diff
@@ -74,16 +74,16 @@ public function load($resource, $locale, $domain = 'messages')
...
+                WHERE `id_lang` = ' . $localeResults[$locale]['id_lang'] . '
+                AND theme ' . ($this->theme !== null ? ' IN ("' . $this->theme->getName() . '"' . ($this->theme->get('parent') ? ',"' . $this->theme->get('parent') . '"' : '') . ')' : 'IS NULL');
 
         $translations = Db::getInstance()->executeS($selectTranslationsQuery) ?: [];
-
+        
         $catalogue = new MessageCatalogue($locale);
         $this->addTranslationsToCatalogue($translations, $catalogue);
-
+
```
Comment (kpodemski) [inline src/PrestaShopBundle/Translation/Loader/SqlTranslationLoader.php]:
```suggestion
```

File: src/PrestaShopBundle/Translation/Loader/SqlTranslationLoader.php [hunk 3]
```diff
@@ -74,16 +74,14 @@ public function load($resource, $locale, $domain = 'messages')
...
-            AND theme ' . ($this->theme !== null ? '= "' . $this->theme->getName() . '"' : 'IS NULL');
+                SELECT `key`, `translation`, `domain`
+                FROM `' . _DB_PREFIX_ . 'translation`
+                WHERE `id_lang` = ' . $localeResults[$locale]['id_lang'] . '
+                AND theme ' . ($this->theme !== null ? ' IN ("' . $this->theme->getName() . '"' . ($this->theme->get('parent') ? ',"' . $this->theme->get('parent') . '"' : '') . ')' : 'IS NULL');
 
         $translations = Db::getInstance()->executeS($selectTranslationsQuery) ?: [];
-
         $catalogue = new MessageCatalogue($locale);
         $this->addTranslationsToCatalogue($translations, $catalogue);
```
Comment (kpodemski) [inline src/PrestaShopBundle/Translation/Loader/SqlTranslationLoader.php:88]:
```suggestion
        $this->addTranslationsToCatalogue($translations, $catalogue);

```

**Review by kpodemski — DISMISSED** [review]

**Review by nicosomb — DISMISSED** [review]

**Review by hibatallahAouadni — COMMENTED** [review]
Hello @ThomasDrd 

I still encounter the issue with your PR, see the attached screen record below:

[pr32359.webm](https://github.com/PrestaShop/PrestaShop/assets/23308427/4dcaae28-7837-45cd-91fb-7cd8d0dd52df)

Please check  and feedback.
Thanks!

**Comment by kpodemski** [comment]
@hibatallahAouadni 

you need to add some custom phrases to your child theme, otherwise you have nothing to translate because translations are coming from the parent theme

**Comment by sallemiines** [comment]
> @hibatallahAouadni
> 
> you need to add some custom phrases to your child theme, otherwise you have nothing to translate because translations are coming from the parent theme

Hello @kpodemski 

Could you please add custom phrases  to the classic child theme and sent it to the QA, so that we can check the PR ! 

Thank you !

**Comment by kpodemski** [comment]
Hello all. This doesn't fix the problem fully if you have multistore and use child themes. I proposed a bit different solution, which should be better as it correctly generates a translations catalog for all the active themes.

https://github.com/PrestaShop/PrestaShop/pull/33133

@ThomasDrd could you let me know if that works for you?

**Review by nicosomb — CHANGES_REQUESTED** [review]
No more changes in your pull request 🤔 I don't think it's normal.

**Pull request closed without merging**